app/scripts/controller-init/messengers/snaps/execution-service-messenger.ts

```diff
@@ -0,0 +1,22 @@
+import { Messenger } from '@metamask/base-controller';
+
+export type ExecutionServiceMessenger = ReturnType<
+  typeof getExecutionServiceMessenger
+>;
+
+/**
+ * Get a restricted messenger for the execution service. This is scoped to the
+ * actions and events that the execution service is allowed to handle.
+ *
+ * @param messenger - The messenger to restrict.
+ * @returns The restricted messenger.
+ */
+export function getExecutionServiceMessenger(
+  messenger: Messenger<never, never>,
+) {
+  return messenger.getRestricted({
```

Comment:
This makes me wonder if the execution service should be passed by reference to the SnapController and not exposed to the messaging system at all 🤔

Reply:
Yeah, but if the execution service doesn't need state and doesn't need to communicate with anyone other than the SnapController, it starts to seem a bit pointless 😄

Reply:
I don't have a strong opinion here, but would prefer changing it in a different PR.

Reply:
Yes, definitely not in scope here